build.gradle

```diff
@@ -1,17 +1,23 @@
 plugins {
-  id 'java-library'
+    id 'java-library'
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
```

Comment:
Change these lines from:
`          // Destination
            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
                authentication(userName: ossrhUsername, password: ossrhPassword)
            }
            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
                authentication(userName: ossrhUsername, password: ossrhPassword)
            }`

To:
`          // Destination
            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
                    authentication(userName: findProperty('ossrhUsername'), password: findProperty('ossrhPassword'))
            }
            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
                authentication(userName: findProperty('ossrhUsername'), password: findProperty('ossrhPassword'))
            }`

Reply:
> Isn't this restricting to partner J8 ? If you remove them, whatever version of JDK is on target (partner) build env will be picked up automatically.
> Build settings should be parameterized, e.x check:
> org.gradle.java.home=/path_to_jdk_directory
> project.hasProperty('javaVer') ? project.getProperty('javaVer') : '1.8'

this explicitly treats our project's minimum SDK as Java 8 .
sourceCompatibility
Defines which language version of Java your source files should be treated as.

targetCompatibility
Defines the minimum JVM version your code should run on, i.e. it determines the version of byte code the compiler generates.

So I think we need this, let me know if you think otherwise